### PR TITLE
feat(SpokePool): Add depositFor public function to override `depositor` address in FundsDeposited event

### PR DESCRIPTION
**Pull request by nicholaspai**
Fixes ACX-1669

# Motivation

Enables deposits sent through third parties (aggregators, SpokePoolVerifier) to be sped up by end user

# Implementation

Logic in `deposit` is copied into internal function `_deposit` that takes one additional param `depositor`.

### CODE REVIEW

File: contracts/SpokePool.sol [hunk 1]
```diff
@@ -906,6 +958,62 @@ abstract contract SpokePool is
...
+        emit FundsDeposited(
+            amount,
+            chainId(),
+            destinationChainId,
+            relayerFeePct,
+            newDepositId,
+            quoteTimestamp,
+            originToken,
+            recipient,
+            depositor,
```
Comment (nicholaspai) [inline contracts/SpokePool.sol:1012]: This is the only line changed, setting `depositor` instead of `msg.sender` to `FundsDeposited.depositor`

**Comment by nicholaspai** [comment]
I added a new minor version bump here so we can more easily tag the latest `deployed` version

File: contracts/SpokePool.sol (hunk 1 above)
Comment (mrice32) [reply contracts/SpokePool.sol:1012]: Yeah, wanted to triple check that the transferFrom lines weren't changed. +1

**Review by mrice32 — APPROVED** [review]
LGTM!

**Pull request merged**